packages/analytics-browser/src/plugins/file-download-tracking.ts

```diff
@@ -0,0 +1,50 @@
+import { BrowserClient, PluginType, Event, EnrichmentPlugin } from '@amplitude/analytics-types';
+import { BrowserConfig } from '../config';
+
+const FILE_DOWNLOAD_EVENT = 'file_download';
+
+export const fileDownloadTracking = (): EnrichmentPlugin => {
+  const name = 'fileDownloadTracking';
+  const type = PluginType.ENRICHMENT;
+  const setup = async (config: BrowserConfig, amplitude?: BrowserClient) => {
+    /* istanbul ignore if */
+    if (!amplitude) {
+      // TODO: Add required minimum version of @amplitude/analytics-browser
+      config.loggerProvider.warn(
+        'File download tracking requires a later version of @amplitude/analytics-browser. File download events are not tracked.',
+      );
+      return;
+    }
+
+    const ext =
+      /\.(pdf|xlsx?|docx?|txt|rtf|csv|exe|key|pp(s|t|tx)|7z|pkg|rar|gz|zip|avi|mov|mp4|mpe?g|wmv|midi?|mp3|wav|wma)$/;
+    const links = Array.from(document.getElementsByTagName('a'));
```

Comment:
For SPA, anchor tag might not be displayed on the page initially. Maybe consider adding an event listener on the document / body and add conditional check to detect if the event target is anchor.

Reply:
Good call out on dynamically added elements! Though adding an event listener to the document/body may be overkill and likely compromise performance. I will look at other options.